db/sql/src/channels.rs

```diff
@@ -161,20 +184,49 @@ impl HoprDbChannelOperations for HoprDb {
             .await
     }
 
-    async fn finish_channel_update<'a>(&'a self, tx: OptTx<'a>, editor: ChannelEditor) -> Result<ChannelEntry> {
+    async fn begin_channel_update<'a>(&'a self, tx: OptTx<'a>, id: &Hash) -> Result<Option<ChannelEditor>> {
+        let id_hex = id.to_hex();
+        self.nest_transaction(tx)
+            .await?
+            .perform(|tx| {
+                Box::pin(async move {
+                    Ok::<_, DbSqlError>(
+                        Channel::find()
+                            .filter(channel::Column::ChannelId.eq(id_hex))
+                            .one(tx.as_ref())
+                            .await?
+                            .map(|model| ChannelEditor {
+                                orig: model.clone().try_into().ok(),
+                                model: model.into_active_model(),
+                                delete: false,
+                            }),
+                    )
+                })
+            })
+            .await
+    }
+
+    async fn finish_channel_update<'a>(&'a self, tx: OptTx<'a>, editor: ChannelEditor) -> Result<Option<ChannelEntry>> {
         let epoch = editor.model.epoch.clone();
-        let parties = ChannelParties(editor.orig.source, editor.orig.destination);
+        let origin = editor.orig.ok_or(DbSqlError::DecodingError)?;
+
+        let parties = ChannelParties(origin.source, origin.destination);
         let ret = self
             .nest_transaction(tx)
             .await?
             .perform(|tx| {
                 Box::pin(async move {
                     if !editor.delete {
                         let model = editor.model.update(tx.as_ref()).await?;
-                        Ok::<_, DbSqlError>(model.try_into()?)
+                        match ChannelEntry::try_from(model) {
+                            Ok(channel) => Ok::<_, DbSqlError>(Some(channel)),
+                            Err(_) => {
+                                Ok::<_, DbSqlError>(None) // If the channel entry cannot be created, return None
+                            }
+                        }
```

Comment:
This does not make much sense imho. If there's an error in the TryFrom, it should bubble up as an error. If the ChannelEditor says the entry is corrupted, there can be a condition that short-circuits by `return Ok(None)` after the update.